litmus-portal/authentication/api/handlers/user_handlers.go

```diff
@@ -0,0 +1,176 @@
+		userResponse, err := service.CreateUser(&userRequest)
+		if err == utils.ErrUserExists {
+			log.Info(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrUserExists], presenter.CreateErrorResponse(utils.ErrUserExists))
+			return
+		}
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		c.JSON(200, userResponse)
+	}
+}
+func UpdateUser(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var userRequest entities.User
+		err := c.BindJSON(&userRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		userResponse, err := service.UpdateUser(&userRequest)
+		if err != nil {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+		}
+		c.JSON(200, userResponse)
+	}
+}
+func FetchUsers(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		users, err := service.GetUsers()
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		c.JSON(200, users)
+	}
+}
+
+func LoginUser(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var userRequest entities.User
+		err := c.BindJSON(&userRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		if userRequest.UserName == "" || userRequest.Password == "" {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		userRequest.UserName = utils.SanitizeString(userRequest.UserName)
+		authenticatedUser, err := service.FindUser(&userRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidCredentials], presenter.CreateErrorResponse(utils.ErrInvalidCredentials))
+			return
+		}
+		token, err := authenticatedUser.GetSignedJWT()
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		expiryTime := time.Duration(utils.JWTExpiryDuration) * 60
+		c.JSON(200, gin.H{
+			"access_token": token,
+			"expires_in":   expiryTime,
+			"type":         "Bearer",
+		})
+	}
+}
+
+func UpdatePassword(service user.Service) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		username := c.MustGet("username").(string)
+		uid := c.MustGet("uid").(string)
+		var adminUser entities.User
+		adminUser.UserName = username
+		var userPasswordRequest entities.UserPassword
+		err := c.BindJSON(&userPasswordRequest)
+		userPasswordRequest.Username = username
+		adminUser.ID, _ = primitive.ObjectIDFromHex(uid)
```

Comment:
Where is adminUser used in this function?

Reply:
Why has this been resolved? What is the use of the `adminUser` variable being created?